core/domain/exp_jobs_one_off.py

```diff
@@ -116,34 +128,54 @@ def reduce(key, values):
         yield (key, values)
 
 
-class MathExpressionInputInteractionOneOffJob(jobs.BaseMapReduceOneOffJobManager):  # pylint: disable=line-too-long
-    """Job that produces a list of (exploration, state) pairs that use the Math
-    Expression Interaction and that have rules that contain [<, >, =] to
-    identify and prevent usage of equation/inequalities.
+class MathExpressionValidationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that produces a list of explorations that use the MathExpressionInput
+    along with the validity and type(expression/equation) of the inputs present
+    in the exploration.
+
+    This validation is done by the 'is_valid_math_expression' or the
+    'is_valid_math_equation' function present in schema_utils.py.
     """
 
+    valid_exps_seen = 0
+
     @classmethod
     def entity_classes_to_map_over(cls):
         return [exp_models.ExplorationModel]
 
     @staticmethod
     def map(item):
+        is_valid_math_expression = schema_utils.get_validator(
+            'is_valid_math_expression')
+        is_valid_math_equation = schema_utils.get_validator(
+            'is_valid_math_equation')
+        ltt = latex2text.LatexNodes2Text()
+
         if not item.deleted:
             exploration = exp_fetchers.get_exploration_from_model(item)
             for state_name, state in exploration.states.items():
                 if state.interaction.id == 'MathExpressionInput':
                     for group in state.interaction.answer_groups:
                         for rule_spec in group.rule_specs:
-                            rule = rule_spec.inputs['x']
-                            if any(sep in rule for sep in SEPARATORS):
+                            rule_input = ltt.latex_to_text(
+                                rule_spec.inputs['x'])
+                            validity = 'Invalid'
+                            if is_valid_math_expression(rule_input):
+                                validity = 'Valid Expression'
+                                MathExpressionValidationOneOffJob.valid_exps_seen += 1  #pylint: disable=line-too-long
```

Comment:
Why are you storing this on the class? Just declare local vars.

Also I feel like this completely misses the point of mapreduce jobs. Please read up a bit more on how these work and look at other examples in the codebase. Any combination of data between individual map threads should be done in reduce(). The whole point of MR is so that things can run on a distributed system so don't assume that the same running instance is processing all of the models. In the map() phase you'll still need to yield everything, let reduce() sort out the combination / filtering / etc.

Reply:
Oh, I got confused. I've localized the variable now, thanks!